### PR TITLE
Pract.01: Add 04-06 | Nikita Beliy

### PR DESCRIPTION
**Pull request by N1k1ta-white**
Nikita Beliy
SI 1 9MI800022

### CODE REVIEW

File: Task04.cpp [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+const int NAME_SIZE_MAX = 30;
+
+struct Student {
+    char name[NAME_SIZE_MAX + 1];
+    uint64_t fac_num;
+};
+
+void initStudent(const Student& student) {
+    cout << "Name: " << student.name << endl
+         << "Fac. number: " << student.fac_num << endl;
```
Comment (khatoni) [inline Task04.cpp]: The name of the function suggest that you will initialise student, instead you are printing it.

File: Task04.cpp [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+}
+
+void str_trans(char str1[], const char str2[]) {
+    int rotator = -1;
+    while (str2[++rotator] != '\0' && rotator != NAME_SIZE_MAX)
+        str1[rotator] = str2[rotator];
+    str1[rotator] = '\0';
+}
+
+void change_info(Student& student, const char name[], int fac_num) {
```
Comment (khatoni) [inline Task04.cpp]: const int fac_num

File: Task05.cpp [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+struct Complex_num {
+    int r, i;
+};
+
+Complex_num sum_complex_num(const Complex_num &a, const Complex_num &b) {
+    Complex_num res{ a.r + b.r, a.i + b.i };
+    return res;
+}
+
+void read_complex_num(const Complex_num &num) {
```
Comment (khatoni) [inline Task05.cpp]: again the name is ambiguous

File: Task06.cpp [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+using std::cout;
+using std::cin;
+using std::endl;
+
+struct Point {
+    int x, y;
+};
+
+void read_point(const Point &p) {
```
Comment (khatoni) [inline Task06.cpp]: I suggest  void print(const Point& p) , because read_point may refer to reading the point from the console

File: Task06.cpp [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+        cout << "IV\n";
+    else if (p.x == 0 && p.y == 0)
+        cout << "Center\n";
+    else if (p.x == 0)
+        cout << "On X-axis\n";
+    else if (p.y == 0)
+        cout << "On Y-axis\n";
+}
+
+bool in_circle(const Point& p, int r) {
```
Comment (khatoni) [inline Task06.cpp]: const int r

**Review by khatoni — CHANGES_REQUESTED** [review]

**Comment by Justsvetoslavov** [comment]
Check example solutions.

**Pull request closed without merging**